nodeJS/authentication/authentication_basics.md

```diff
@@ -153,20 +157,17 @@ We need to add 3 functions to our app.js file, and then add an app.post for our
 #### Function one : setting up the LocalStrategy
 ~~~javascript
 passport.use(
-  new LocalStrategy((username, password, done) => {
-    User.findOne({ username: username }, (err, user) => {
-      if (err) { 
-        return done(err);
-      }
-      if (!user) {
-        return done(null, false, { message: "Incorrect username" });
-      }
+  new LocalStrategy(async(username, password, done) => {
+    try {
+      const user = await User.findOne({ username: username });
       if (user.password !== password) {
         return done(null, false, { message: "Incorrect password" });
-      }
+      };
       return done(null, user);
-    });
-  })
+    } catch(err) {
+      return done(null, false, { message: "Incorrect username" });
+    };
+  });
```

Comment:
Please add a line to account for if `user` is falsy. You are catching an error in line 168, not a null user. The catch block should be identical to line 159 in the original, because we want to send the error along to our error handler.

Reply:
Thank you for clarifying that; I'll add the user falsy case back in.